notebooks/kaggle/store-sales-time-series-forecasting/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,1515 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "corr_coefficients = (train_data_copy.corr().where(np.triu(np.ones(train_data_copy.corr().shape), k=1).astype(bool))\n",
+    "                  .stack()\n",
+    "                  .sort_values(ascending=False))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "corr_coefficients[:30]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "corr_coefficients[:-10:-1]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>The highest correlation coefficients refer to pairs like 'state' - 'city' or 'event_type' - 'description' or 'locale_name' - 'description', which don't give us any new information. As we will see later, all information about holidays is excess, because the correlation between a binary feature indicating the existence of a holiday on a particular day and all other features is very close to zero.</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "train_data_copy = train_data.copy().drop(['event_type', 'locale', 'locale_name', 'description', 'state'], axis=1)\n",
+    "train_data_copy['transferred'] = np.where((train_data_copy['transferred'] == True), 1, train_data_copy['transferred'])\n",
+    "train_data_copy['transferred'] = np.where((train_data_copy['transferred'] == False), 1, train_data_copy['transferred'])\n",
+    "train_data_copy['transferred'] = train_data_copy['transferred'].fillna(0)\n",
+    "train_data_copy['transferred'].value_counts()\n",
+    "\n",
+    "is_holiday = pd.get_dummies(train_data_copy['transferred'])\n",
+    "train_data_copy = pd.concat((is_holiday, train_data_copy), axis=1)\n",
+    "train_data_copy = train_data_copy.rename(columns={1: \"is_holiday\"})\n",
+    "train_data_copy = train_data_copy.drop([0, 'transferred'], axis=1)\n",
+    "\n",
+    "train_data_copy['transactions'] = train_data_copy['transactions'].fillna(0)\n",
+    "\n",
+    "train_data_copy = pd.get_dummies(train_data_copy)\n",
+    "\n",
+    "corr = train_data_copy.corr()\n",
+    "plt.figure(figsize=(30, 25))\n",
+    "\n",
+    "sns.heatmap(corr[(corr >= 0.3) | (corr <= -0.3)], \n",
+    "            cmap='viridis', vmax=1.0, vmin=-1.0, linewidths=0.1,\n",
+    "            annot=True, annot_kws={\"size\": 8}, square=True);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>That is the prove: there is no correlation between 'is_holiday' feature and other features. It means that <b>THERE IS NO REASON TO SAVE ALL THE FEATURES, WHICH DESCRIBE HOLIDAYS AND EVENTS</b> - their influence on target and other features is very close to 0.</i>"
```

Comment:
> THERE IS NO REASON TO SAVE ALL THE FEATURES, WHICH DESCRIBE HOLIDAYS AND EVENTS
For linear models - probably yes, but for more complex model I would not be so categorical.